contracts/interfaces/IPeriphery.sol

```diff
@@ -8,11 +8,71 @@ import "../utils/CustomErrors.sol";
 import "../interfaces/IWETH.sol";
 
 interface IPeriphery is CustomErrors {
-    // events
+    // structs for events
+    struct PositionDetails {
+        IMarginEngine marginEngine;
+        address owner;
+        int24 tickLower;
+        int24 tickUpper;
+    }
+
+    struct SwapOutput {
+        int256 variableTokenDelta;
+        uint256 cumulativeFeeIncurred;
+        int256 marginRequirement;
+    }
 
+    // events
     /// @dev emitted after new lp margin cap is set
     event MarginCap(IVAMM vamm, int256 lpMarginCapNew);
 
+    /// @dev emitted when new lp margin cumulative is set
+    event MarginCumulative(IVAMM vamm, int256 lpMarginCumulative);
+
+    /// @dev emitted when position is settled and all margin withdrawn
+    event SettlePositionAndWithdrawMargin(PositionDetails position);
```

Comment:
when we settle a position and withdraw its margin can we provide more data here like cashflows, etc, not sure if anything is returned to the periphery from the core tho

Reply:
``settlePosition`` function in core does not return anything :(